File: Groove.SP/groove-sp-web/src/app/features/vessel/models/vessel.model.ts
import { VesselStatus } from "src/app/core";

export class VesselModel {
    id: number;
    name: string;
    code: string;
    isRealVessel: boolean;
    status: VesselStatus

    constructor() {
        this.status = VesselStatus.Active;
        this.isRealVessel = true;
    }
}

File: Groove.CSFE/Groove.CSFE.Core/Entities/VesselModel.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Groove.CSFE.Core.Entities
{
    public class VesselModel : Entity
    {
        public long Id { set; get; }
        public string Code { set; get; }
        public string Name { set; get; }
        public VesselStatus Status { get; set; }
        public bool IsRealVessel { get; set; }

    }
}
